src/test/java/org/takes/servlet/RqFromTest.java

```diff
@@ -0,0 +1,161 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.servlet;
+
+import com.google.common.base.Joiner;
+import java.io.IOException;
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.takes.rq.RqFake;
+import org.takes.rq.RqPrint;
+
+/**
+ * Test case for {@link  RqFrom}.
+ *
+ * @since 1.15
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class RqFromTest {
+    @Test
+    public void method() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't add a method to a servlet request",
+            new RqPrint(
+                new RqFrom(
+                    new HttpServletRequestFake(
+                        new RqFake()
+                    )
+                )
+            ).printHead(),
+            Matchers.startsWith("GET /")
+        );
+    }
+
+    @Test
+    public void header() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't add a header to a servlet request",
+            new RqPrint(
+                new RqFrom(
+                    new HttpServletRequestFake(
+                        new RqFake(
+                            new ListOf<>(
+                                "GET /a-test",
+                                "foo: bar"
+                            ),
+                            ""
+                        )
+                    )
+                )
+            ).printHead(),
+            Matchers.startsWith(
+                Joiner.on("\r\n").join(
+                    "GET /a-test",
+                    "Host: localhost",
+                    "foo: bar",
+                    "X-Takes-LocalAddress: 127.0.0.1",
+                    "X-Takes-RemoteAddress: 127.0.0.1"
+                )
+            )
+        );
+    }
+
+    @Test
+    public void hostInHeader() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't set a host in a servlet request",
+            new RqPrint(
+                new RqFrom(
+                    new HttpServletRequestFake(
+                        new RqFake(
+                            new ListOf<>(
+                                "GET /one-more-test",
+                                "Host: www.thesite.com"
+                            ),
+                            ""
+                        )
+                    )
+                )
+            ).printHead(),
+            Matchers.startsWith(
+                Joiner.on("\r\n").join(
```

Comment:
Why not use `org.cactoos.text.JoinedText`? (Example: https://github.com/yegor256/takes/blob/master/src/test/java/org/takes/tk/TkHtmlTest.java)